src/common/AudioStreamBuilder.cpp

```diff
@@ -156,7 +157,20 @@ Result AudioStreamBuilder::openStream(AudioStream **streamPP) {
         }
     }
 
-    result = streamP->open(); // TODO review API
+    // If MMAP has a problem in this case then disable it temporarily.
+    bool wasMMapOriginallyEnabled = AAudioExtensions::getInstance().isMMapEnabled();
+    bool wasMMapTemporarilyDisabled = false;
+    if (wasMMapOriginallyEnabled) {
+        bool isMMapSafe = QuirksManager::getInstance().isMMapSafe(childBuilder);
+        if (!isMMapSafe) {
+            AAudioExtensions::getInstance().setMMapEnabled(false);
+            wasMMapTemporarilyDisabled = true;
```

Comment:
We only re-enabled MMap if `isMMapSafe` == false - is that right? I feel like there's either an error, or some context around this quirk which I'm missing.

Reply:
The code should disable MMAP if it is not safe in this situation.

Is there a better word than "safe"? Maybe "WorkingProperly" or "OK"?

Reply:
MMAPEnabled only has effect during the open call! That was probably not clear.

Reply:
I added a comment.